packages/library/src/rpc-integer-overflow-error.ts

```diff
@@ -1,45 +1,45 @@
-    readonly value: bigint;
-    constructor(methodName: string, keyPath: KeyPath, value: bigint) {
-        let argumentLabel = '';
-        if (typeof keyPath[0] === 'number') {
-            const argPosition = keyPath[0] + 1;
-            const lastDigit = argPosition % 10;
-            const lastTwoDigits = argPosition % 100;
-            if (lastDigit == 1 && lastTwoDigits != 11) {
-                argumentLabel = argPosition + 'st';
-            } else if (lastDigit == 2 && lastTwoDigits != 12) {
-                argumentLabel = argPosition + 'nd';
-            } else if (lastDigit == 3 && lastTwoDigits != 13) {
-                argumentLabel = argPosition + 'rd';
-            } else {
-                argumentLabel = argPosition + 'th';
-            }
+export function createSolanaJsonRpcIntegerOverflowError(
+    methodName: string,
+    keyPath: KeyPath,
+    value: bigint,
+): SolanaError<typeof SOLANA_ERROR__RPC_INTEGER_OVERFLOW> {
+    let argumentLabel = '';
+    if (typeof keyPath[0] === 'number') {
+        const argPosition = keyPath[0] + 1;
+        const lastDigit = argPosition % 10;
+        const lastTwoDigits = argPosition % 100;
+        if (lastDigit == 1 && lastTwoDigits != 11) {
+            argumentLabel = argPosition + 'st';
+        } else if (lastDigit == 2 && lastTwoDigits != 12) {
+            argumentLabel = argPosition + 'nd';
+        } else if (lastDigit == 3 && lastTwoDigits != 13) {
+            argumentLabel = argPosition + 'rd';
         } else {
-            argumentLabel = `\`${keyPath[0].toString()}\``;
+            argumentLabel = argPosition + 'th';
         }
-        const path =
-            keyPath.length > 1
-                ? keyPath
-                      .slice(1)
-                      .map(pathPart => (typeof pathPart === 'number' ? `[${pathPart}]` : pathPart))
-                      .join('.')
-                : null;
-        super(
-            `The ${argumentLabel} argument to the \`${methodName}\` RPC method` +
-                `${path ? ` at path \`${path}\`` : ''} was \`${value}\`. This number is ` +
-                'unsafe for use with the Solana JSON-RPC because it exceeds ' +
-                '`Number.MAX_SAFE_INTEGER`.',
-        );
-        this.keyPath = keyPath;
-        this.methodName = methodName;
-        this.value = value;
+    } else {
+        argumentLabel = `\`${keyPath[0].toString()}\``;
     }
-    get name() {
-        return 'SolanaJsonRpcIntegerOverflowError';
+    const path =
+        keyPath.length > 1
+            ? keyPath
+                  .slice(1)
+                  .map(pathPart => (typeof pathPart === 'number' ? `[${pathPart}]` : pathPart))
+                  .join('.')
+            : undefined;
+    const error = new SolanaError(SOLANA_ERROR__RPC_INTEGER_OVERFLOW, {
+        argumentLabel,
+        keyPath: keyPath as readonly (string | number | symbol)[],
+        methodName,
+        optionalPathLabel: path ? ` at path \`${path}\`` : '',
```

Comment:
So I guess that's how we go around some of the more complex error messages without using callbacks. 👍

Reply:
Yeah, I'm super sad about pushing this error-formatting code into the _production_ bundle but I think this error is important enough to collect this amount of detail, _particularly_ in production, where it will occur most insidiously.